utilities/fault_injection_fs.h

```diff
@@ -584,31 +714,28 @@ class FaultInjectionTestFS : public FileSystemWrapper {
     int frames;
     ErrorType type;
     bool retryable;
+    bool has_data_loss;
 
     explicit ErrorContext(uint32_t seed)
         : rand(seed),
           enable_error_injection(false),
           callstack(nullptr),
           frames(0),
-          retryable(false) {}
+          retryable(false),
+          has_data_loss(false) {}
     ~ErrorContext() {
       if (callstack) {
         free(callstack);
       }
     }
   };
 
-  std::unique_ptr<ThreadLocalPtr> thread_local_error_;
-  bool enable_write_error_injection_;
-  bool enable_metadata_write_error_injection_;
-  Random write_error_rand_;
-  int write_error_one_in_;
-  int metadata_write_error_one_in_;
-  std::atomic<int> read_error_one_in_;
-  bool inject_for_all_file_types_;
-  std::vector<FileType> write_error_allowed_types_;
-  // File types where direct writable is skipped.
   std::set<FileType> direct_writable_types_;
+  std::set<Env::IOActivity> io_activties_exempted_from_fault_injection;
+  std::unique_ptr<ThreadLocalPtr> injected_thread_local_read_error_;
```

Comment:
Any reason these are unique_ptr vs. just direct members?

Reply:
No - fixed - I didn't give much of a thought when using the old implementation